### ISSUE TITLE
NVDA does not read the level in multilevel nested lists

### ISSUE THREAD

**Issue by TanjaKleut** [nvaccess/nvda#8103]
Dear all,

NVDA announces the list and the number of items in the list, but it does not read the level of the list item, e.g. if you want to have a list where level 1 is parent, level 2 are the children, level 3 grandchildren, etc.
NVDA does not read this information in none of the browsers that I tried: Mozilla Firefox, Google Chrome, Microsoft Edge.

Examples with code
1. https://codepen.io/bernardoantunes/pen/inrbh?q=organogram&order=popularity&depth=everything&show_for

2. https://codepen.io/Sambo7/pen/QwyyxP?q=organisation%20chart&order=popularity&depth=everything&show_forks=false 

3. https://codepen.io/evelyn415/pen/PZbLzX?q=organogram&order=popularity&depth=everything&show_forks=false

4. https://codepen.io/siiron/pen/aLkdE?q=organization%20chart&order=popularity&depth=everything&show_forks=false 

The same issue happens on the NVDA user guide
https://www.nvaccess.org/files/nvda/documentation/userGuide.html

Under NVDA > Preferences > Documents the user can indicate whether he/she wants to be announced of lists, but it would be good to have the possibility to read the level of items in multilevel lists.

I used:
- NVDA 2018.1
- Mozilla Firefox 59.0.1
- Google Chrome 64.0.3282.18
- Microsoft Edge 41.16299.248.0

**Comment by Qchristensen** [comment]
Possibly related to (or perhaps the work to fix this could also fix) #6815 since list level is not announced in Word either.

**Comment by Adriani90** [comment]
Related to #7804. Please follow up the discussions there because it is also connected to the "Pua bullets" pull request referenced in there.
cc: @feerrenrut, @dkager

**Comment by jiri-thiemel-te** [comment]
I'm just a developer who uses NVDA for testing, but I thought there would be some more advanced ways of navigating nested lists. And it looks like the don't exist. It's beoynd me how the users can reasonably navigate inside them (e.g. inside a complex navigation where multiple levels are displayed at once, like sitemap). (I'd be really curious to hear if anyone can comment on that. 🙏)

If the current options are in deed not sufficient, I do have some ideas, based just on my expectations and lack of experience with NVDA:

- How including lists in the Elements List, somehow. Display their label if they have one, otherwise the would have to be described somehow else, e.g. "List with 8 items (first item: Example item)". The tree would get expanded exactly on the item from which it was open from.
- Or some way to display just the whole current list as a tree.
  - Either as a completely new type of dialog with a new command to open it,
  - Or as part of the Elements List where a new option would appear called e.g. "Current List" when this dialog is open from a list item.

These are just a wild ideas, something I'd expect as a user to speed up my navigation and also something I don't find that complicated from implementation perspective as it should extend the current options in a consistent and compatible way. But opinions of users who depend on screen readers matter the most, just trying to help.

**Comment by SaschaCowley** [comment]
@jiri-thiemel-te [said](https://github.com/nvaccess/nvda/issues/8103#issuecomment-2435663303)
> It's beoynd me how the users can reasonably navigate inside them (e.g. inside a complex navigation where multiple levels are displayed at once, like sitemap). (I'd be really curious to hear if anyone can comment on that. 🙏)

I don't typically have too much trouble. I tend to make heavy use of the move to end/start of container commands (`comma` and `shift+comma`) when interacting with nested lists.

**Comment by XLTechie** [comment]
I don't have a problem navigating them, but I admit it can be a trial knowing which level you're on, which can sometimes be important.

I don't know what a better accessibility pattern might be though.

**Comment by Adriani90** [comment]
I agree this can be improved in NVDA when using web pages in browsers. For now you can also use vertical navigation to jump between levels, I assigned nvda+alt+up and down arrows in the input gestures dialog for that feature. However, it doesn't work 100% reliabele but it gives a rough idea of how the list should be perceived, I tested on the first codepen example in the description.
Note that you already can use object navigation to detect the list level, this seems to be reported by NVDA properly when using the navigator.

Especially in complex lists such as organigrams it would be a huge help to have kind of a tree view element, similar to what we have e.g. in Windows file explorer.
But before that, #8193 needs to be implemented I guess.

In my view, container navigation via comma and shift+comma is too complicated, because it jumps out of the list level and not at the end or at the beginning of it. If you press shift+comma you don't jump at the beginning of the list level, you were before, but to a list level that is far above, especially when the list has many items. See #15732 for more details on the drawbacks of the current container navigation design in NVDA.
